app/controllers/assignments_controller.rb

```diff
@@ -85,13 +85,23 @@ def index
     @chatroom = Chatroom.find(params[:course_id])
     authorize @chatroom
     @message = Message.new
+    @messages = @chatroom.messages
+    new_messages = @messages.where.not("user_id = #{current_user.id}")
```

Comment:
@Jimreed91 

So, `view_at `of all of messages which are accessed by the receiver will be updated, correct?
After the conversation in the last week, I feel it might be better to have the col as a `boolean` flag just to show the read status because if we update `view_at` every time, it would consume more computer power changing more data records. More data we have huge set of messages, more computer power would be required.

When we change the column with boolean type (for example, `read`), then we can have

`new_messages = @messages.where.not(user_id: #{current_user.id}).and(read: nil)
`

Then set true for `read`.
And using symbol in where clause would bring about better practice.

Reply:
This makes sense, definitely should have the `read: nil` condition. Re. The column being `datetime`/`boolean` I'm thinking since it defaults to `nil` on a new message we could keep it as `datetime` (calling `message.viewed_at?` on a message will return Boolean based on whether it has been assigned a date time yet).

 I like this approach because we might want to show a user when their message was seen, let me know what you think. In the case we go this way I'd maybe change the col name to 'viewed' so we can call 'viewed?'to get the Boolean.

Not sure what I was thinking using string over a symbol there, will refactor and maybe later move the logic out of the controller also

Reply:
It looks like i was incorrect about active record returning a boolean for a different column type (it does work in console but i guess ruby is converting it)